src/spec.c

```diff
@@ -2802,9 +2964,17 @@ int IndexSpec_UpdateDoc(IndexSpec *spec, RedisModuleCtx *ctx, RedisModuleString
     return REDISMODULE_ERR;
   }
 
+  QueryError status = {0};
+
+  if(spec->scan_failed_OOM) {
+    QueryError_SetError(&status, QUERY_INDEXBGOOMFAIL, "Index background scan did not complete due to OOM. New documents will not be indexed.");
```

Comment:
```suggestion
    QueryError_SetError(&status, QUERY_INDEXBGOOMFAIL, "Index background scan did not complete due to OOM. Index may contain only partial data.");
```
Nitpick: This error message sounds more appropriate as we are not setting the index as faulty anymore, WDYT?

Reply:
Will change in all version

Reply:
It's to notify that the index will not receive new docs (I understood that's what should happen).
If that's not the case I'll fix that

Reply:
What if the memory limit increased after a while, and some will try to insert a document synchronously via HSET?